nntrainer/layers/time_dist.cpp

```diff
@@ -60,7 +60,79 @@ int TimeDistLayer::initialize(Manager &manager) {
 }
 
 void TimeDistLayer::forwarding(bool training) {
-  // NYI
+  Tensor &hidden_ = net_hidden[0]->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+
+  Tensor hidden_g, h_g;
+
+  TensorDim ho_dim = hidden_.getDim();
+  TensorDim in_dim = input_.getDim();
+
+  // input_.dim = [ 1, b, h, w ] : nntrainer only support 1,2,3 transpose
+  input_.reshape({1, in_dim[0], in_dim[2], in_dim[3]});
```

Comment:
i think this will have to `getBatchSlixe()` or `getSharedDataTensor()` as reshape requires the new shape and old shape to be of equal size, which is not the case here.

Reply:
I don't understand, how this is reducing the size and line 75 goes back to the original size?

Reply:
the channel is 1. so.. the order of date is same between [ b, 1, h, w ] and [ 1, b, h, w ].  Therefore we just do the reshape for that. and Tensor only support C, H, W transpose. ( Now, It B, H, W ). Now Transpose ( [ 1, H, B, W] ) and reshape it to [ H, 1, B, W ].

Reply:
thanks for the explanation.